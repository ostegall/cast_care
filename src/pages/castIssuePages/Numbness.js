import { Text, View, Pressable, Linking } from 'react-native';
import styles from '../../styles';
let phone = " (984)-974-5700 ";
let phone_url = "tel://+19849745700";

export default function Numbness({ navigation }) {
  //Solutions for user that has numbness in cast
  return (
    <View style={{ flex: 1, paddingTop: 20 }}>
      <Text style={styles.title_text}>
        Problem Description:
      </Text>
      <Text style={styles.page_text}>
        Are there any areas of numbness? Does patient's limb have a sleepy feeling or pins and needles that is not alleviated by elevation?
      </Text>
      <Text style={styles.title_text}>
        Why this is a Problem:
      </Text>
      <Text style={styles.page_text}>
        If there is numbness, this could indicate that the cast/splint is pinching or rubbing on a nerve, and it needs to be addressed.
      </Text>
      <Text style={styles.title_text}>
        Correcting the Problem:
      </Text>
      <Text style={styles.page_text}>
        Elevate the extremity to see if the numbness goes away. If the numbness does not go away, schedule an appointment.
      </Text>
      <Text style={styles.title_text}>
        Who to Contact:
      </Text>
      <Text style={styles.page_text}>
        Follow-up with the office by calling 
        <Text style={styles.url_text} //separate text box for an inline link to call UNC hospital
          onPress={() => Linking.openURL(phone_url)}>
          {phone}
        </Text>
        to schedule a same-day appointment or come into the Orthonow walk-in clinic. If patient is not in the NC Triangle area, please go to your local ED.
      </Text>
      <View style={{ alignItems: 'center', padding: 20 }}>
        <Pressable style={styles.button}
          onPress={() => {
            navigation.navigate("Info") //Navigates to page for finding nearest hospital
          }}
        >
          <Text style={styles.button_text}>
            Find your nearest location
          </Text>
        </Pressable>
      </View>
    </View>
  )
}