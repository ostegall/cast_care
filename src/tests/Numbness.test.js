import renderer from 'react-test-renderer';
import { shallow } from 'enzyme'

import Numbness from '../pages/castIssuePages/Numbness';

//Makes sure app component renders correctly without crashing
it("renders without crashing", () => {
  shallow(<Numbness />);
});

//Snapshot test to ensure UI stays consistent
it('renders correctly', () => {
  const tree = renderer.create(<Numbness />).toJSON();
  expect(tree).toMatchSnapshot();
});