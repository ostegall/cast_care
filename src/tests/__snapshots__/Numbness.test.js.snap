// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingTop": 20,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 23,
        "fontWeight": "bold",
        "letterSpacing": 0.5,
        "textAlign": "center",
      }
    }
  >
    Problem Description:
  </Text>
  <Text
    style={
      Object {
        "alignItems": "center",
        "fontSize": 17,
        "letterSpacing": 0.5,
        "marginBottom": 5,
        "marginLeft": 20,
        "marginRight": 20,
        "marginTop": 5,
      }
    }
  >
    Are there any areas of numbness? Does patient's limb have a sleepy feeling or pins and needles that is not alleviated by elevation?
  </Text>
  <Text
    style={
      Object {
        "fontSize": 23,
        "fontWeight": "bold",
        "letterSpacing": 0.5,
        "textAlign": "center",
      }
    }
  >
    Why this is a Problem:
  </Text>
  <Text
    style={
      Object {
        "alignItems": "center",
        "fontSize": 17,
        "letterSpacing": 0.5,
        "marginBottom": 5,
        "marginLeft": 20,
        "marginRight": 20,
        "marginTop": 5,
      }
    }
  >
    If there is numbness, this could indicate that the cast/splint is pinching or rubbing on a nerve, and it needs to be addressed.
  </Text>
  <Text
    style={
      Object {
        "fontSize": 23,
        "fontWeight": "bold",
        "letterSpacing": 0.5,
        "textAlign": "center",
      }
    }
  >
    Correcting the Problem:
  </Text>
  <Text
    style={
      Object {
        "alignItems": "center",
        "fontSize": 17,
        "letterSpacing": 0.5,
        "marginBottom": 5,
        "marginLeft": 20,
        "marginRight": 20,
        "marginTop": 5,
      }
    }
  >
    Elevate the extremity to see if the numbness goes away. If the numbness does not go away, schedule an appointment.
  </Text>
  <Text
    style={
      Object {
        "fontSize": 23,
        "fontWeight": "bold",
        "letterSpacing": 0.5,
        "textAlign": "center",
      }
    }
  >
    Who to Contact:
  </Text>
  <Text
    style={
      Object {
        "alignItems": "center",
        "fontSize": 17,
        "letterSpacing": 0.5,
        "marginBottom": 5,
        "marginLeft": 20,
        "marginRight": 20,
        "marginTop": 5,
      }
    }
  >
    Follow-up with the office by calling
    <Text
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "color": "blue",
          "fontSize": 17,
          "letterSpacing": 0.5,
          "marginBottom": 5,
          "marginLeft": 20,
          "marginRight": 20,
          "marginTop": 5,
        }
      }
    >
       (984)-974-5700 
    </Text>
    to schedule a same-day appointment or come into the Orthonow walk-in clinic. If patient is not in the NC Triangle area, please go to your local ED.
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "padding": 20,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#4292c7",
          "borderRadius": 10,
          "elevation": 2,
          "flexDirection": "row",
          "height": 80,
          "justifyContent": "center",
          "minWidth": 88,
          "paddingLeft": 16,
          "paddingRight": 16,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.35,
          "shadowRadius": 5,
          "width": 120,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 14,
            "letterSpacing": 0.25,
            "textAlign": "center",
          }
        }
      >
        Find your nearest location
      </Text>
    </View>
  </View>
</View>
`;
